### PR TITLE
chore(permission): Add minimal permission for litmus-admin rbac

### PR DESCRIPTION
**Pull request by uditgaurav**
Signed-off-by: Udit Gaurav <udit.gaurav@mayadata.io>

## Proposed changes

Summarize your changes here to communicate with the maintainers and make sure to put the link of that issue

## Types of changes

What types of changes does your code introduce to Litmus? Put an `x` in the boxes that apply
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices applies)

## Checklist

Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.
- [ ] I have read the [CONTRIBUTING](https://github.com/litmuschaos/litmus/blob/master/CONTRIBUTING.md) doc
- [ ] I have signed the commit for DCO to be passed.
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ ] I have added necessary documentation (if appropriate)

## Dependency
- Please add the links to the dependent PR need to be merged before this (if any).

### CODE REVIEW

File: docs/litmus-admin-rbac.yaml [hunk 1]
```diff
@@ -14,9 +14,12 @@ metadata:
...
     name: litmus-admin
 rules:
-- apiGroups: ["","apps","batch","extensions","litmuschaos.io","openebs.io","storage.k8s.io"]
-  resources: ["chaosengines","chaosexperiments","chaosresults","cstorpools","cstorvolumereplicas","configmaps","secrets","pods","pods/exec","pods/log","pods/eviction","jobs","replicasets","deployments","daemonsets","statefulsets","persistentvolumeclaims","persistentvolumes","storageclasses","services","events"]
+- apiGroups: ["","apps","batch","extensions","litmuschaos.io"]
+  resources: ["pods","pods/exec","pods/eviction","jobs","daemonsets","events","chaosresults","chaosengines"]
   verbs: ["create","delete","get","list","patch","update", "deletecollection"]
+- apiGroups: ["","apps","litmuschaos.io"]
+  resources: ["pods","configmaps","secrets","events","services","chaosexperiments","pods/log","replicasets","deployments","statefulsets",services]
+  verbs: ["get","list","patch"]
```
Comment (ksatchit) [inline docs/litmus-admin-rbac.yaml:22]: can we separate out resources needing patch? For ex: statefulsets, replicasets etc., we don't patch today

**Comment by uditgaurav** [comment]
A pipeline run with the changes: https://gitlab.mayadata.io/litmuschaos/litmus-e2e/pipelines/12828

File: docs/litmus-admin-rbac.yaml [hunk 2]
```diff
@@ -14,9 +14,12 @@ metadata:
   labels:
     name: litmus-admin
 rules:
-- apiGroups: ["","apps","batch","extensions","litmuschaos.io","openebs.io","storage.k8s.io"]
-  resources: ["chaosengines","chaosexperiments","chaosresults","cstorpools","cstorvolumereplicas","configmaps","secrets","pods","pods/exec","pods/log","pods/eviction","jobs","replicasets","deployments","daemonsets","statefulsets","persistentvolumeclaims","persistentvolumes","storageclasses","services","events"]
+- apiGroups: ["","apps","batch","extensions","litmuschaos.io"]
+  resources: ["pods","pods/exec","pods/eviction","jobs","daemonsets","events","chaosresults","chaosengines"]
   verbs: ["create","delete","get","list","patch","update", "deletecollection"]
+- apiGroups: ["","apps","litmuschaos.io"]
```
Comment (sumitnagal) [inline docs/litmus-admin-rbac.yaml:20]: I used below for my pod experiment ```- apiGroups: ["","litmuschaos.io","batch","apps"]
  resources: ["pods","deployments","events","jobs","configmaps","chaosengines","chaosexperiments","chaosresults","pods/log"]
  verbs: ["create","list","get","patch","update","delete"]
- apiGroups: [""]
  resources: ["nodes"]
  verbs : ["get","list"]```

File: docs/litmus-admin-rbac.yaml [hunk 3]
```diff
@@ -14,9 +14,12 @@ metadata:
...
+  resources: ["pods","configmaps","secrets","events","services","chaosexperiments","pods/log","replicasets","deployments","statefulsets",services]
```
Comment (sumitnagal) [inline docs/litmus-admin-rbac.yaml:21]: you can remove pods and events from here

File: docs/litmus-admin-rbac.yaml [hunk 4]
```diff
@@ -14,9 +14,12 @@ metadata:
...
+  verbs: ["get","list","patch","update"]
 - apiGroups: [""]
   resources: ["nodes"]
```
Comment (sumitnagal) [inline docs/litmus-admin-rbac.yaml:24]: why we need patch and update on nodes

Comment (uditgaurav) [reply docs/litmus-admin-rbac.yaml:24]: Experiments like Node Taint needs that 

ref: https://github.com/litmuschaos/litmus-go/blob/0fa4a508f9c67241dbb28d3b60da1588582bcfcc/chaoslib/litmus/node-taint/lib/node-taint.go#L198

File: docs/litmus-admin-rbac.yaml (hunk 3 above)
Comment (uditgaurav) [reply docs/litmus-admin-rbac.yaml:21]: Yes Thanks !

File: docs/litmus-admin-rbac.yaml (hunk 2 above)
Comment (uditgaurav) [reply docs/litmus-admin-rbac.yaml:20]: Yes, this is fine. 

We may need statefulsets and ds as well.

`"configmaps","secrets","services","chaosexperiments","pods/log","replicasets","deployments","statefulsets",services`

^^ These don't need the create and delete permissions

**Review by ksatchit — APPROVED** [review]

**Review by ksatchit — APPROVED** [review]

**Pull request merged**